includes/Block/Web_Stories_Block.php

```diff
@@ -0,0 +1,364 @@
+						'type'    => 'string',
+						'default' => __( 'Web Story', 'web-stories' ),
+					],
+					'poster'           => [
+						'type' => 'string',
+					],
+					'width'            => [
+						'type'    => 'number',
+						'default' => 360,
+					],
+					'height'           => [
+						'type'    => 'number',
+						'default' => 600,
+					],
+					'align'            => [
+						'type'    => 'string',
+						'default' => 'none',
+					],
+					'stories'          => [
+						'type'    => 'array',
+						'default' => [],
+					],
+					'viewType'         => [
+						'type'    => 'string',
+						'default' => '',
+					],
+					'numOfStories'     => [
+						'type'    => 'number',
+						'default' => 5,
+					],
+					'numOfColumns'     => [
+						'type'    => 'number',
+						'default' => 2,
+					],
+					'sizeOfCircles'    => [
+						'type'    => 'number',
+						'default' => 150,
+					],
+					'orderByValue'     => [
+						'type'    => 'string',
+						'default' => '',
+					],
+					'isShowingTitle'   => [
+						'type'    => 'boolean',
+						'default' => true,
+					],
+					'isShowingExcerpt' => [
+						'type'    => 'boolean',
+						'default' => false,
+					],
+					'isShowingDate'    => [
+						'type'    => 'boolean',
+						'default' => false,
+					],
+					'isShowingAuthor'  => [
+						'type'    => 'boolean',
+						'default' => false,
+					],
+					'isShowingViewAll' => [
+						'type'    => 'boolean',
+						'default' => false,
+					],
+					'viewAllLinkLabel' => [
+						'type'    => 'string',
+						'default' => '',
+					],
+					'authors'          => [
+						'type'    => 'array',
+						'default' => [],
+					],
+					'imageOnRight'     => [
+						'type'    => 'boolean',
+						'default' => false,
+					],
+					'isStyleSquared'   => [
+						'type'    => 'boolean',
+						'default' => false,
+					],
+				],
+				'render_callback' => [ $this, 'render_block' ],
+				'editor_script'   => self::SCRIPT_HANDLE,
+				'editor_style'    => self::SCRIPT_HANDLE,
+			]
+		);
+	}
+
+	/**
+	 * Returns script settings.
+	 *
+	 * @since
+	 *
+	 * @return array Script settings.
+	 */
+	private function get_script_settings() {
+		$rest_base = Story_Post_Type::POST_TYPE_SLUG;
+
+		return [
+			'publicPath' => WEBSTORIES_PLUGIN_DIR_URL . 'assets/js/',
+			'authors'    => $this->get_story_authors(),
+			'config'     => [
+				'maxNumOfStories' => $this->max_num_of_stories,
+				'api'             => [
+					'stories' => sprintf( '/web-stories/v1/%s', $rest_base ),
+				],
+			],
+		];
+	}
+
+	/**
+	 * Get a list of story authors
+	 *
+	 * @since
+	 *
+	 * @return array Author list
+	 */
+	protected function get_story_authors() {
+		// @see Roles listed in \Google\Web_Stories\Story_Post_Type::add_caps_to_roles() method
+		return get_users(
+			[
+				'role__in' => [ 'administrator', 'editor', 'author', 'contributor' ],
+				'orderby'  => 'display_name',
+				'number'   => -1,
+				'fields'   => [ 'ID', 'display_name' ],
+			]
+		);
+	}
```

Comment:
Please remove. Sites with 10K users, it would mean that the editor would not load at all. Please do this as a REST API call using -1 param there.

Reply:
Agree with you, I'm working on refactoring as well as restructuring some of these components.